plugins/modules/virtual_disk_attach.py

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+module: virtual_disk_attach
+
+author:
+  - Polona Mihalič (@PolonaM)
+short_description: Clones an uploaded virtual disk and attaches it to a virtual machine.
+description:
+  - Clones an uploaded virtual disk and attaches it to a virtual machine.
+  - If selected slot is not empty, selected virtual disk will not be attached.
+version_added: 1.2.0
+extends_documentation_fragment:
+  - scale_computing.hypercore.cluster_instance
+  - scale_computing.hypercore.vm_name
+seealso:
+  - module: scale_computing.hypercore.virtual_disk
+  - module: scale_computing.hypercore.virtual_disk_info
+  - module: scale_computing.hypercore.vm_disk
+options:
+  name:
+    description:
+      - Virtual disk name that we want to attach.
+      - Used as a unique identifier of an uploaded virtual disk.  # should we use UUID??
+    type: str
+    required: true
+  block_device: # in vm_disk this is named "items", should we use disk?
```

Comment:
I would like to have similar naming as vm and vm_disk modules. vm_disk: `items` does not fit (plural), and `item` is also ugly. vm: `disks` is plural => you win, `disk` looks like best name to me.

Reply:
Updated to "disk"